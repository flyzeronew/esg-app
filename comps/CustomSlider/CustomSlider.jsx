import Slider from 'react-slick';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import styles from './CustomSlider.module.css';
import classNames from 'classnames/bind';
const cx = classNames.bind(styles);
function CustomSlider({ settings, children, customClass }) {
  return (
    <div role="slider" className={cx('CustomSlider', customClass ? cx(customClass) : '')}>
      <Slider {...settings}>{children}</Slider>
    </div>
  );
}
export default CustomSlider;
