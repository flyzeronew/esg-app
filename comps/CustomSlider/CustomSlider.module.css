.CustomSlider  :global(.slick-dots){
    position:absolute;
    bottom:40px;
    left: 40px;
    text-align: left;
}
 .CustomSlider :global(.slick-dots li){
    width: 40px;
    height: 6px;
    margin: 0;
    margin-right: 7px;
    overflow: hidden;
    padding:0;
}
 .CustomSlider :global(.slick-dots li button ){
    width: 100%;
    height: 100%;
    cursor: pointer;
    color: transparent;
    border: 0;
    outline: none;
    background: transparent;
    overflow: hidden;
    padding:0;
}
/* custom classes from components starts */
.viewList :global(.slick-dots li){
    width: 30px;
}
/* Ends */
 .CustomSlider :global(.slick-dots li button:before){
    width: 100%;
    height: 100%;
    background-color: #D4D4D4; 
    font-size: 0;
    content:'';
    opacity: 1;
    border-radius:30px;
}
.CustomSlider :global(.slick-dots li.slick-active button:before) {
    background-color:red;
  } 
@media (max-width:1279px) {
    .CustomSlider :global(.slick-dots li){
       width: 34px;
   }
   .viewList :global(.slick-dots li){
    width: 30px;
}
} 