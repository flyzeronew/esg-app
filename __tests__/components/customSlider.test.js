import { render } from '@testing-library/react';
import CustomSlider from '@/comps/CustomSlider/CustomSlider';

const settings = {
    dots: true,
    fade: true,
    infinite: true,
    speed: 500,
    slidesToShow: 1,
    slidesToScroll: 1,
    swipeToSlide:false,
    autoplay: true,
    autoplaySpeed:5000, 
};
describe("Custom slider snapshot", () => {
    it('renders slider unchanged', () => {
        const {container} = render(<CustomSlider settings={settings} ></CustomSlider>);
        expect(container).toMatchSnapshot();
    });
    it('checks for custom classes', () => {
        const renderedCustomSlider = render(<CustomSlider settings={settings} customClass={"tipsView"}></CustomSlider>);
        expect(renderedCustomSlider.getByRole("slider")).toHaveClass('tipsView');
    });
});

