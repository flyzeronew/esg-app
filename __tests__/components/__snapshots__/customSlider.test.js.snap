// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom slider snapshot renders slider unchanged 1`] = `
<div>
  <div
    class="CustomSlider"
    role="slider"
  >
    <div
      class="slick-slider slick-initialized"
      dir="ltr"
    >
      <button
        class="slick-arrow slick-prev"
        data-role="none"
        style="display: block;"
        type="button"
      >
         
        Previous
      </button>
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1;"
        />
      </div>
      <button
        class="slick-arrow slick-next"
        data-role="none"
        style="display: block;"
        type="button"
      >
         
        Next
      </button>
    </div>
  </div>
</div>
`;
